packages/as-types/assembly/serializable.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Implement this interface so that you can use `Args` to serialize/de-serialize your objects.
+ *
+ * @example
+ * ```typescript
+ * const args = new Args().addObject(new Address(input));
+ * const deserialized = args.nextObject<Address>(new Address());
+ * ```
+ */
+export interface Serializable<T> {
+  /**
+   * @returns the bytes
+   */
+  serialize(): StaticArray<u8>;
+
+  /**
+   * @param data - the serialized data
+   * @param offset - current offset
+   * @returns the length of the deserialized object
```

Comment:
Are you sure that this is not the new offset instead? If not, I think you should change the description to the number of bytes read from the data buffer.